source/extensions/filters/network/common/redis/codec.h

```diff
@@ -25,18 +28,85 @@ enum class RespType { Null, SimpleString, BulkString, Integer, Error, Array };
 class RespValue {
 public:
   RespValue() : type_(RespType::Null) {}
-  ~RespValue() { cleanup(); }
 
-  RespValue(const RespValue& other);             // copy constructor
-  RespValue& operator=(const RespValue& other);  // copy assignment
-  bool operator==(const RespValue& other) const; // test for equality, unit tests
+  RespValue(std::shared_ptr<RespValue> base_array, const RespValue& command, const uint64_t start,
+            const uint64_t end)
+      : type_(RespType::CompositeArray) {
+    new (&composite_array_) CompositeArray(std::move(base_array), command, start, end);
+  }
+  virtual ~RespValue() { cleanup(); }
+
+  RespValue(const RespValue& other);                // copy constructor
+  RespValue(RespValue&& other) noexcept;            // move constructor
+  RespValue& operator=(const RespValue& other);     // copy assignment
+  RespValue& operator=(RespValue&& other) noexcept; // move assignment
+  bool operator==(const RespValue& other) const;    // test for equality, unit tests
   bool operator!=(const RespValue& other) const { return !(*this == other); }
 
   /**
    * Convert a RESP value to a string for debugging purposes.
    */
   std::string toString() const;
 
+  class CompositeArray {
+  public:
+    CompositeArray() = default;
```

Comment:
this is needed for the type() method for RespValue and the recreate() method for SplitCommand.  It can be remove after the other change when I remove the recreate() method.

Reply:
This is needed for the type() method and the recreate() in command splitter.  I think it can be removed with #8606 when we remove the recreate() method.  Personally I'd like to make all the RespValue const outside of the decoder, but that's a bigger change.